File: Java/day29.java
import java.math.*;
import java.util.*;

public class day29
{
    public static void main(String[] args) {
        // Write your code here
        Scanner sc = new Scanner(System.in);
        boolean funny = true;
        int T = sc.nextInt();
        for (int i = 0; i < T; i++) {
            String raw_str = sc.next();

            char[] str = raw_str.toCharArray();
            StringBuilder reversed_str_1 = new StringBuilder(raw_str);
            String reversed_str_2 = new String(reversed_str_1.reverse());
            char[] reversed_str = reversed_str_2.toCharArray();

            for (int j = 1; j < reversed_str.length; j++) {
                if (Math.abs((int)str[j] - (int)str[j-1]) !=
                    Math.abs((int)reversed_str[j] - (int)reversed_str[j-1]))
                    funny = false;
            }

            if (funny)
                System.out.println("Funny");
            else {
                System.out.println("Not Funny");
                funny = true;
            }
        }
    }
}

File: Scala/day29.scala
object Solution {
  def main(args: Array[String]) {
    // Write your code here
    val sc = new java.util.Scanner(System.in)
    var funny = true
    val T = sc.nextInt()
    for (i <- 1 to T) {
      val raw_str = sc.next()
      val reversed_str = raw_str.reverse
      for (j <- 1 to raw_str.length()-1) {
        if (Math.abs(raw_str(j).toInt - raw_str(j-1).toInt) !=
          Math.abs(reversed_str(j).toInt - reversed_str(j-1).toInt))
          funny = false
      }
      if (funny) println("Funny") else { println("Not Funny"); funny = true }
    }
  }
}
